RELEASES.md

```diff
@@ -77,18 +77,30 @@ Support horizons will be defined corresponding to a release branch, identified
 by `<major>.<minor>`. Releases branches will be in one of several states:
 
 - __*Next*__: The next planned release branch.
-- __*Active*__: The release branch is currently supported and accepting patches.
+- __*Active*__: The release is a stable branch which is currently supported and accepting patches.
 - __*Extended*__: The release branch is only accepting security patches.
+- __*LTS*__: The release is a long term stable branch which is currently supported and accepting patches.
 - __*End of Life*__: The release branch is no longer supported and no new patches will be accepted.
 
-Releases will be supported up to one year after a _minor_ release. This means that
+Releases will be supported at least one year after a _minor_ release. This means that
 we will accept bug reports and backports to release branches until the end of
 life date. If no new _minor_ release has been made, that release will be
 considered supported until 6 months after the next _minor_ is released or one year,
 whichever is longer. Additionally, releases may have an extended security support
 period after the end of the active period to accept security backports. This
 timeframe will be decided by maintainers before the end of the active status.
 
+Long term stable (_LTS_) releases will be supported for at least three years after
+their initial _minor_ release. These branches will accept bug reports and
+backports until the end of life date. They may also accept a wider range of
+patches than non-_LTS_ releases to support the longer term maintainability of the
+branch, including library dependency, toolchain (including Go) and other version updates
+which are needed to ensure each release is built with fully supported dependencies and
+remains usable by containerd clients. There should be at least a 6 month overlap between
```

Comment:
> remains usable by containerd clients

this may create an interesting challenge with k8s versions that are out of OSS support. When backporting changes for the active k8s versions, old k8s versions compatibility with containerd may degrade. I'd suggest to phrase something like:

```suggestion
remains usable by supported containerd clients. Note, LTS version of containerd may stop working for the containerd clients which are not up to date - you need to seek support from these clients authors. There should be at least a 6 month overlap between
```

Reply:
Also in the table in the next section "Kubernetes Support", the list of recommended k8s versions for 1.6 will be fluid now. Bu 2025 recommendation will be different. I think this needs to be specified

Reply:
> old k8s versions compatibility with containerd may degrade

Not our intention.

k8s versions, and container runtimes, out of OSS support are already a cloud provider... non-cve bugs / issues that will not be back ported to n-3 for k8s, and same for container runtimes at EOL, for example.

Wrt. CRI, this change will allow us to back port certain selected non-breaking type changes and extend the span of versions of k8s we can support where the api changes for CRI do not include breaking changes.  As an aside breaking changes should require a CRI version change and thus won't, as a practice, be back ported to a prior api version, we currently support 2 versions of CRI and I don't remember off hand if any changes have been breaking.  If a breaking change does occur to a version of the api and we have a need to back port it we can work together to figure out a way to disable the breaking behavior for older versions of k8s.. e.g. config switch, identify kubelet version/pod spec as a switch, or some other suitable mechanism.

Obvious statement of the day: Optional apis that are not called by an older version of k8s and optional fields not used by an older version of k8s won't break an older version of k8s. 

As you mentioned, we keep a table for k8s support, and if necessary we can expand that table to have a level of detail for optional k8s cri features showing where said cri feature gates/apis may be configured in kubelet and/or containerd..

IOW do the right thing, exercise discretion and care, but have some flexibility.  Practice is showing many cloud providers and other container runtime clients, are moving much slower than k8s's quarterly cycle.

the text in RELEASES.md is being left somewhat broad to leave discretion to the maintainers..

Reply:
My main point is that in the table of suggested k8s versions there should be a note saying that the list will change over time as we cannot suggest that older (out of support) k8s versions is something containerd recommends to use with containerd 1.6

Reply:
Agree in principle from a k/k project support view and not sure we are 100% in sync just yet wrt. containerd recommendation statements in view of this new LTS idea :-) I'll make sure to flag the right k8s people, including you, when we update the containerd k8s support table as we are, of course, integration partners on that statement.  We'll definitely want to be clear about what's been tested, what hasn't been tested, the EOL status wrt the k8s versions that are officially EOL/out of support, and possibly where to go for out of support release help, at least in a conceptual way... 

k8s versions which are officially out of support by k8s may still be in use, and supported by cloud/platform providers... containerd should continue to run some tests for each branch pr/release even against out of support versions of k8s / CRI, even if just to verify a prior version of the CRI api dropped in a newer version of k8s but not yet removed in the containerd branch.  

Having a long term release that goes on as much as 2.5years beyond kubernetes support will definitely expose the support gap more than today's six months to year or so gap between the length of time an old version of containerd is supported and k8s support by the n-3 rule.

Cheers.

Reply:
Y'all can figure out the clearest way to convey K8s support in a follow up or when we add the next K8s release into the list.